feat(scheduler-targets-alpha): `KinesisDataFirehosePutRecord` Target

go-to-k opened the pull request:
This PR adds KinesisDataFirehosePutRecord as a target for an EventBridge scheduler.

Closes #27450.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation reviewed (changes requested):
The pull request linter has failed. See the aws-cdk-automation comment below for failure reasons. If you believe this pull request should receive an exemption, please comment and provide a justification.


A comment requesting an exemption should contain the text `Exemption Request`. Additionally, if clarification is needed add `Clarification Request` to a comment.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: f2ca64d720799d8f229dbab4996f57e2fb7638ea
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bb2b635f-bdae-45a2-90b7-d4014459d1a3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

<b>PRs must pass status checks before we can provide a meaningful review.</b>

If you would like to request an exemption from the status checks or clarification on feedback, please leave a comment on this PR containing `Exemption Request` and/or `Clarification Request`.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: f4435eb86f8592977b990f78db4c84dd237706c4
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c1e3c02a-55c1-4b96-b8ea-29ef72b4671b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9547abad6148ffa7b1efbd039ef1b16fdb66bd09
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=99951ef1-0253-4a57-b961-955a908262e5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 18c0b018f99dff34560d35a6a80c7227321b0651
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f5def948-3ae6-4fde-abfa-4960382047b9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 26930d45376e4b2d81fbe6ff0defd6cd13d77407
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=26377af1-e1e9-4f3d-87e0-a1124c0574db%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 38cf37459a99a39285ccd2cf5a4c4ef4f172653f
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b0efca09-9f1c-4e4c-a918-68e6b50c393d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: fb5b02768b4c7121c93792790f9a6e8c1c8da2a8
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9614567c-8f1b-4962-92e6-cd57a88a623c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: c1214bd95b76a0217b5c41c86ce5bfcff4cd02fb
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ef3baa09-1b81-46fa-979b-b54f3abf7927%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 2cb5cd6fd5a30050ef9f6eb1870929eeb3c580c4
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=9162d17d-f7a6-482e-8b50-188edd7a20f8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

lpizzinidev commented on packages/@aws-cdk/aws-scheduler-targets-alpha/README.md:
```suggestion
Use the `KinesisDataFirehosePutRecord` target to put a record to a Kinesis Data Firehose delivery stream.
```

lpizzinidev commented on packages/@aws-cdk/aws-scheduler-targets-alpha/README.md:
```suggestion
import * as firehose from 'aws-cdk-lib/aws-kinesisfirehose';

declare const deliveryStream: firehose.CfnDeliveryStream;
```

lpizzinidev commented on packages/@aws-cdk/aws-scheduler-targets-alpha/lib/kinesis-data-firehose-put-record.ts:
```suggestion
      throw new Error(`Cannot assign the Firehose delivery stream in region ${this.deliveryStream.stack.region} to the schedule ${Names.nodeUniqueId(schedule.node)} in region ${schedule.env.region}. Both the schedule and the Firehose delivery stream must be in the same region.`);
```

lpizzinidev commented on packages/@aws-cdk/aws-scheduler-targets-alpha/lib/kinesis-data-firehose-put-record.ts:
```suggestion
      throw new Error(`Cannot assign the Firehose delivery stream in account ${this.deliveryStream.stack.account} to the schedule ${Names.nodeUniqueId(schedule.node)} in account ${schedule.env.region}. Both the schedule and the Firehose delivery stream must be in the same account.`);
```

lpizzinidev reviewed (changes requested):
Looks good 👍
Just some minor documentation adjustments needed.

go-to-k commented:
@lpizzinidev 

Thanks! Please check the changes.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 550ddbe980a68009d09d1b14079f312ffe6a461b
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=669bfd30-ba76-4d48-baa0-566d51e88085%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

lpizzinidev reviewed (approved):
Thanks 👍

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: c6b15636a478451420e6057b63c2b09be153885d
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fd1765cb-89c7-4489-a7cf-dea764ed57ac%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: bbb78fd53ecbe787f0428a214a3d288671cf43d2
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1503e21a-efb5-49cf-932d-68930f18166e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9490651fb8bc5123e88aff01178db64939ad6c64
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e185b2d4-0f45-4549-ad37-5951363d2147%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

vinayak-kukreja reviewed (approved):
Hey @go-to-k, thank you for this contribution.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 45a3ee961c300a67b1ca3d5d1a8c6f91d41622f4
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=c1a8dac5-0c27-49bc-8924-bf7b2f1f3aed%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

go-to-k commented:
@vinayak-kukreja 

Thanks for your approval! I fixed conflicts. Please approve again.

I fixed [the other PR](https://github.com/aws/aws-cdk/pull/27845) the same way, but when one of them gets merged, the other one gets conflicts again. If that happens, I will fix it again.

vinayak-kukreja commented:
Hey @go-to-k , thank you for fixing the conflicts. The other PR is now merged. Please let me know once you have fixed conflicts for this PR.

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

go-to-k commented:
@vinayak-kukreja 

Thanks, I fixed new conflicts in this PR again, please look at this.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0fba10931500ec8172de13014ac1461ec0c88595
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1fb11d49-7bc2-4da9-993f-f3c54cb09f41%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: f8d1f44dfb11be9aa46f992bb43b72e3cfc0c4fc
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=457f959d-641a-4f64-8180-b071165de8b2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
